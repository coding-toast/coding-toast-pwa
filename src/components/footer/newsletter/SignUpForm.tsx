import * as React from 'react'
import SignUpButton from '../../buttons/button-signup/SignUpButton'
import NewsletterStyles from './Newsletter.module.scss'

const SignUpForm: React.FC = () => {
  return (
    <form action="">
      <input type="email" name="email" className={NewsletterStyles.emailInput} />
      <SignUpButton />
    </form>
  )
}

export default SignUpForm
