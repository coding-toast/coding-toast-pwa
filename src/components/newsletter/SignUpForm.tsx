import * as React from 'react'
import { Form, Button, Card, Row, Col } from 'react-bootstrap'
import { Link } from 'gatsby'
import { Formik } from 'formik'
import * as Yup from 'yup'

const SignUpForm: React.FC = () => {
  const schema = Yup.object({
    email: Yup.string().email('Invalid email address').required('Required'),
    acceptedTerms: Yup.boolean().required('Required').oneOf([true], 'You must accept the terms and conditions.')
  })

  return (
    <Row xs={1} lg={2} className="justify-content-lg-center my-4">
      <Col>
        <Card body style={{ backgroundColor: 'rgba(255, 255, 255, 0.05)' }}>
          <h2 className="mb-4">Never miss a beat with our Newsletter!</h2>
          <Formik
            initialValues={{ email: '', acceptedTerms: true }}
            validationSchema={schema}
            onSubmit={(values, { setSubmitting }) => {
              setTimeout(() => {
                alert(JSON.stringify(values, null, 2))
                setSubmitting(false)
              }, 400)
            }}
          >
            {({ handleSubmit, handleChange, handleBlur, values, touched, isValid, errors }) => (
              <Form noValidate onSubmit={handleSubmit}>
                <Form.Group controlId="email">
                  <Form.Label>Email address</Form.Label>
                  <Form.Control
                    type="email"
                    placeholder="Enter email"
                    value={values.email}
                    onChange={handleChange}
                    onBlur={handleBlur}
                    isInvalid={touched.email && !isValid}
                    isValid={touched.email && isValid}
                  />
                  <Form.Control.Feedback type="invalid">{errors.email}</Form.Control.Feedback>
                  <Form.Text className="text-muted">We'll never share your email with anyone else.</Form.Text>
                </Form.Group>
                <Form.Group controlId="acceptedTerms">
                  <Form.Check type="checkbox">
                    <Form.Check.Input
                      type="checkbox"
                      checked={values.acceptedTerms}
                      onChange={handleChange}
                      isInvalid={touched.acceptedTerms && !isValid}
                    />
                    <Form.Check.Label className="small text-muted">
                      I have read and agree to the <Link to="/">Terms and Conditions</Link> and <Link to="/">Privacy Policy</Link>
                    </Form.Check.Label>
                    <Form.Control.Feedback type="invalid">{errors.acceptedTerms}</Form.Control.Feedback>
                  </Form.Check>
                </Form.Group>
                <Button type="submit" className="btn-accent-secondary" block>
                  Sign Up
                </Button>
              </Form>
            )}
          </Formik>
        </Card>
      </Col>
    </Row>
  )
}

export default SignUpForm
