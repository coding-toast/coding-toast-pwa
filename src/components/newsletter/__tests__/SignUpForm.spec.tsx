import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignUpForm from '../SignUpForm';

describe('<SignUpForm />', () => {
  it('renders correctly', () => {
    const component = shallow(<SignUpForm />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
