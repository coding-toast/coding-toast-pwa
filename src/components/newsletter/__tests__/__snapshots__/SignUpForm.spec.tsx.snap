// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> renders correctly 1`] = `
<Row
  className="justify-content-lg-center my-4"
  lg={2}
  noGutters={false}
  xs={1}
>
  <Col>
    <Card
      body={true}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.05)",
        }
      }
    >
      <h2
        className="mb-4"
      >
        Never miss a beat with our Newsletter!
      </h2>
      <p
        className="m-1 h4"
      >
        We email a couple of times a month.
      </p>
      <p
        className="m-1 h4"
      >
        No spam.
      </p>
      <p
        className="m-1 mb-4 h4"
      >
        Just the best developer tips.
      </p>
      <Formik
        initialValues={
          Object {
            "acceptedTerms": true,
            "email": "",
          }
        }
        onSubmit={[Function]}
        validationSchema={
          ObjectSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_defaultDefault": [Function],
            "_deps": Array [],
            "_excludedEdges": Array [],
            "_exclusive": Object {},
            "_mutate": undefined,
            "_nodes": Array [
              "acceptedTerms",
              "email",
            ],
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "object",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "fields": Object {
              "acceptedTerms": BooleanSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "boolean",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {
                    true,
                  },
                  "refs": Map {},
                },
                "_whitelistError": [Function],
                "tests": Array [
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
                "type": "boolean",
              },
              "email": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "email": false,
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
                "type": "string",
              },
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
            "type": "object",
          }
        }
      >
        <Component />
      </Formik>
    </Card>
  </Col>
</Row>
`;
